### PR TITLE
Remove app-scroll-position

### PR DESCRIPTION
**Pull request by blasten**

### CODE REVIEW

File: templates/pesto/src/recipe-app.html [hunk 1]
```diff
@@ -131,9 +130,6 @@
         </app-header-layout>
       </app-drawer>
 
-      <app-scroll-position
-          selected="[[_pageData.page]]"
-          reset="[[_isDetailPage(_pageData.page)]]"></app-scroll-position>
```
Comment (frankiefu) [inline templates/pesto/src/recipe-app.html]: Let's add some JS code to save the scroll position so if you navigate back to the list page you are at the previous scroll position.  This code will serve as a sample for users for replacing `app-scroll-position` element.

File: templates/publishing/src/blog-app.html [hunk 2]
```diff
@@ -29,7 +29,6 @@
 <link rel="import" href="../bower_components/app-layout/app-header/app-header.html">
 <link rel="import" href="../bower_components/app-layout/app-header-layout/app-header-layout.html">
 <link rel="import" href="../bower_components/app-layout/app-toolbar/app-toolbar.html">
-<link rel="import" href="../bower_components/app-layout/app-scroll-position/app-scroll-position.html">
```
Comment (frankiefu) [inline templates/publishing/src/blog-app.html]: Let's remove this old [`<app-scrollpos-control>`](https://github.com/PolymerElements/app-layout/blob/4dd4304851cc3b32588b5ecc6015697bf6305f73/templates/publishing/src/blog-app.html#L168-L170) too.

**Comment by blasten** [comment]
PTAL

File: templates/pesto/src/recipe-app.html [hunk 3]
```diff
@@ -118,21 +126,28 @@
 
           <!-- nav menu -->
           <paper-listbox
-              selected="{{_pageData.page}}"
+              selected="[[_pageData.page]]"
```
Comment (keanulee) [inline templates/pesto/src/recipe-app.html]: I understand why there needs to be a separate `_selectedPage` property, but why does this need to change to one-way bound `_pageData.page` with `<a>` links for navigation? This is an issue if a user clicks on the `<paper-item>` outside of the `<a>` (where the paper-item gets selected, but the link isn't clicked).

Comment (blasten) [inline templates/pesto/src/recipe-app.html]: there's a bug in Polymer 2.0 that causes a notification to be missed.

**Comment by MeTaNoV** [comment]
interesting... any other way to have a more flexible implementation?

File: templates/pesto/src/recipe-app.html (hunk 3 above)
Comment (frankiefu) [inline templates/pesto/src/recipe-app.html]: Let's add a comment with a TODO that we should change this back once the issue in Polymer2 is fixed.

**Comment by frankiefu** [comment]
@MeTaNoV We feel that there isn't a guaranteed way to make this element works reliably because to save the current scroll position it needs to do that before the page visibility changes.  Ideally it should be handled inside `iron-pages`.  We will look into providing a better solution in the future.  For now you will need to save/restore scroll position in code.  An example: https://github.com/PolymerElements/app-layout/blob/757d1fd87105c33f5876cfab72d5f9fb6f270244/templates/pesto/src/recipe-app.html#L238-L250

**Comment by frankiefu** [comment]
@blasten Just need to add the [comment](https://github.com/PolymerElements/app-layout/pull/417#pullrequestreview-24871661) and otherwise LGTM

**Comment by MeTaNoV** [comment]
thx @frankiefu for your answer, I agree that it should be handled in any <XXX-pages> since there is not only `iron-pages` but others (including my own `iron-swipeable-pages`) that need to handle this. Do you think that a behaviour could be extracted to be used by the different elements or by extending a base `iron-pages` we could reuse the implementation? I am not yet familiar with 2.0 so I cannot tell directly...
@timvdlippe might be probably interested also!

**Comment by frankiefu** [comment]
@MeTaNoV Yeah, a behavior makes a lot of sense.  Good suggestion!

File: templates/pesto/src/recipe-app.html (hunk 3 above)
Comment (keanulee) [inline templates/pesto/src/recipe-app.html]: Is there a tracking issue in polymer?

Comment (blasten) [inline templates/pesto/src/recipe-app.html]: will file the bug asap.

Comment (blasten) [inline templates/pesto/src/recipe-app.html]: done tracking issue is: https://github.com/polymer/polymer/issues/4405

File: templates/pesto/src/recipe-app.html (hunk 1 above)
Comment (blasten) [inline templates/pesto/src/recipe-app.html]: done

File: templates/publishing/src/blog-app.html (hunk 2 above)
Comment (blasten) [inline templates/publishing/src/blog-app.html]: done

**Pull request merged**